lib/publiccloud/k8s_provider.pm

```diff
@@ -0,0 +1,70 @@
+# SUSE's openQA tests
+#
+# Copyright 2021 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Helper class for amazon connection and authentication
+#
+# Maintainer: qa-c team <qa-c@suse.de>
+
+package publiccloud::k8s_provider;
+use Mojo::Base -base;
+use testapi;
+use mmapi 'get_current_job_id';
+
+
+has region => undef;
+has resource_name => sub { get_var('PUBLIC_CLOUD_RESOURCE_NAME', 'openqa-vm') };
+has container_registry => sub { get_var("PUBLIC_CLOUD_CONTAINER_IMAGES_REGISTRY", 'suse-qec-testing') };
+has provider_client => undef;
+
+sub init {
+    my ($self, $service) = @_;
+    die('The service must be specified') if (!$service);
+
+    my $provider = get_required_var('PUBLIC_CLOUD_PROVIDER');
+
+    if ($provider eq 'EC2') {
```

Comment:
I would condsider this as not oop design. The Parent should _not_ know what childs it has.
I would keep this client creation in the corresponding child classes